workspaces/src/operations.rs

```diff
@@ -52,16 +56,22 @@ impl<'a> Function<'a> {
     /// Similiar to `args`, specify an argument that is JSON serializable and can be
     /// accepted by the equivalent contract. Recommend to use something like
     /// `serde_json::json!` macro to easily serialize the arguments.
-    pub fn args_json<U: serde::Serialize>(mut self, args: U) -> anyhow::Result<Self> {
-        self.args = serde_json::to_vec(&args)?;
-        Ok(self)
+    pub fn args_json<U: serde::Serialize>(mut self, args: U) -> Self {
```

Comment:
it doesn't make much sense to me that this method would not return a result anymore and keep the error until the call is executed. It seems to me that it should be either:
- Keep the signature as is, do serialization when args_x is called
- Make `Function` generic on the args type, serialize when function is executed

What is the motivation for having this change included with the error handling?

Reply:
That's how reqwest works: 

https://docs.rs/reqwest/latest/src/reqwest/async_impl/request.rs.html#441-457

This seems like a nice API, from the caller's perspective -- that extra `?` would be useless like 99.999% of the time, and you'd have to type it for every `.json`. 

The builder here is kinda equivalent to calling a single function with a whole bunch of paramters. As you'd have a single `?` when calling the fn, there's no need to pessimize the builder. 

We do need to make sure that we use this pattern of infailable builder consistently .

Reply:
okay, I'll concede this, I suppose if keeping this pattern, introducing a new generic for the JSON type probably isn't ideal.

There is one thing that can change which is changing the inner type of `Function` to be a `Result` instead of keeping redundant data and performing operations when the serialization has already failed. This is optional but is what `request` does in the link you provided also. Might make things more maintainable in the future?